google-cloud-spanner/src/test/java/com/google/cloud/spanner/OpenTelemetryBuiltInMetricsTracerTest.java

```diff
@@ -0,0 +1,293 @@
+        SimulatedExecutionTime.ofException(Status.UNAVAILABLE.asRuntimeException()));
+
+    // Execute a simple read/write transaction using only mutations. This will use the
+    // BeginTransaction RPC to start the transaction. That RPC will first return UNAVAILABLE, then
+    // be retried by Gax, and succeed. The retry should show up in the tracing.
+    client.write(ImmutableList.of(Mutation.newInsertBuilder("foo").set("bar").to(1L).build()));
+
+    stopwatch.elapsed(TimeUnit.MILLISECONDS);
+
+    Attributes expectedAttributesBeginTransactionOK =
+        expectedBaseAttributes
+            .toBuilder()
+            .put(BuiltInMetricsConstant.STATUS_KEY, "OK")
+            .put(BuiltInMetricsConstant.METHOD_KEY, "Spanner.BeginTransaction")
+            .build();
+
+    Attributes expectedAttributesBeginTransactionFailed =
+        expectedBaseAttributes
+            .toBuilder()
+            .put(BuiltInMetricsConstant.STATUS_KEY, "UNAVAILABLE")
+            .put(BuiltInMetricsConstant.METHOD_KEY, "Spanner.BeginTransaction")
+            .build();
+
+    MetricData attemptCountMetricData =
+        getMetricData(metricReader, BuiltInMetricsConstant.ATTEMPT_COUNT_NAME);
+    assertThat(getAggregatedValue(attemptCountMetricData, expectedAttributesBeginTransactionOK))
+        .isEqualTo(1);
+    // Attempt count should have a failed metric point for Begin Transaction.
+    assertThat(getAggregatedValue(attemptCountMetricData, expectedAttributesBeginTransactionFailed))
+        .isEqualTo(1);
+
+    MetricData operationCountMetricData =
+        getMetricData(metricReader, BuiltInMetricsConstant.OPERATION_COUNT_NAME);
+    assertThat(getAggregatedValue(operationCountMetricData, expectedAttributesBeginTransactionOK))
+        .isEqualTo(1);
+    // Operation count should not have a failed metric point for Begin Transaction as overall
+    // operation is success..
+    assertThat(
+            getAggregatedValue(operationCountMetricData, expectedAttributesBeginTransactionFailed))
+        .isEqualTo(0);
+  }
+
+  private MetricData getMetricData(InMemoryMetricReader reader, String metricName) {
+    String fullMetricName = BuiltInMetricsConstant.METER_NAME + "/" + metricName;
+    Collection<MetricData> allMetricData = Collections.emptyList();
+
+    // Fetch the MetricData with retries
+    for (int attemptsLeft = 10; attemptsLeft > 0; attemptsLeft--) {
+      allMetricData = reader.collectAllMetrics();
+      List<MetricData> matchingMetadata =
+          allMetricData.stream()
+              .filter(md -> md.getName().equals(fullMetricName))
+              .collect(Collectors.toList());
+      assertWithMessage(
+              "Found multiple MetricData with the same name: %s, in: %s",
+              fullMetricName, matchingMetadata)
+          .that(matchingMetadata.size())
+          .isAtMost(1);
+
+      if (!matchingMetadata.isEmpty()) {
+        return matchingMetadata.get(0);
+      }
+
+      try {
+        Thread.sleep(100);
```

Comment:
Could we reduce this to 1 millisecond and increase the number of attempts to 1000? That should reduce the overall waiting time.